src/core/toga/widgets/canvas.py

```diff
@@ -1,329 +1,1137 @@
+        translate = Translate(tx, ty)
+        self.__add_drawing_object(translate)
+        return translate
 
-    # Drawing Paths
+    def reset_transform(self):
+        """Constructs and returns a :class:`ResetTransform <ResetTransform>`.
 
-    @contextmanager
-    def fill(self, fill_rule='nonzero', preserve=False):
-        """Fills the subpaths with the current fill style
+        Returns:
+            :class:`ResetTransform <ResetTransform>` object.
 
-        A drawing operator that fills the current path according to the current
-        fill rule, (each sub-path is implicitly closed before being filled).
+        """
+        reset_transform = ResetTransform()
+        self.__add_drawing_object(reset_transform)
+        return reset_transform
+
+    ###########################################################################
+    # Text drawing
+    ###########################################################################
+
+    def write_text(self, text, x=0, y=0, font=None):
+        """Constructs and returns a :class:`WriteText <WriteText>`.
+
+        Writes a given text at the given (x,y) position. If no font is provided,
+        then it will use the font assigned to the Canvas Widget, if it exists,
+        or use the default font if there is no font assigned.
 
         Args:
-            fill_rule (str): 'nonzero' is the non-zero winding rule and
-                             'evenodd' is the even-odd winding rule
-            preserve (bool): Preserves the path within the Context
+            text (string): The text to fill.
+            x (float, optional): The x coordinate of the text. Default to 0.
+            y (float, optional): The y coordinate of the text. Default to 0.
+            font (:class:`toga.Font`, optional): The font to write with.
+
+        Returns:
+            :class:`WriteText <WriteText>` object.
 
         """
-        self._impl.new_path()
-        yield
-        if fill_rule is 'evenodd':
-            self._impl.fill(fill_rule, preserve)
+        write_text = WriteText(text, x, y, font)
+        self.__add_drawing_object(write_text)
+        return write_text
+
+
+class Canvas(InterfaceMixin, Widget):
+    """Create new canvas
+
+    Args:
+        id (str):  An identifier for this widget.
+        style (:obj:`Style`): An optional style object. If no
+            style is provided then a new one will be created for the widget.
+        factory (:obj:`module`): A python module that is capable to return a
+            implementation of this class with the same name. (optional &
+            normally not needed)
+    """
+
+    def __init__(self, id=None, style=None, factory=None):
+        super().__init__(id=id, style=style, factory=factory)
+        self._canvas = self
+        self.drawing_objects = []
+
+        # Create a platform specific implementation of Canvas
+        self._impl = self.factory.Canvas(interface=self)
+
+        self._impl.context(self)  # Create root context
```

Comment:
This seems like it should be an adjective (e.g., `set_context` or `create_context`); as a noun, it seem odd to be invoked but not have the result stored.